truss/truss_config.py

```diff
@@ -582,7 +582,7 @@ def obj_to_dict(obj, verbose: bool = False):
                 )
             elif isinstance(field_curr_value, TRTLLMConfiguration):
                 d["trt_llm"] = transform_optional(
-                    field_curr_value, lambda data: data.model_dump(mode="json")
+                    field_curr_value, lambda data: data.dict()
```

Comment:
is it gauranteed that the dict is json serializable?

Reply:
Yes, all (sub)fields of `TRTLLMConfiguration` are either primitive or pydantic models. Also the pre-change code would raise an exception if something is not JSON-serializable.